File: src/main/java/org/moinex/model/dto/InvestmentPerformanceDTO.java
/*
 * Filename: InvestmentPerformanceDTO.java
 * Created on: February 17, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.model.dto;

import java.math.BigDecimal;
import java.time.YearMonth;
import java.util.Map;

/**
 * Data Transfer Object to hold investment performance metrics
 */
public record InvestmentPerformanceDTO(
        Map<YearMonth, BigDecimal> monthlyInvested,
        Map<YearMonth, BigDecimal> portfolioValues,
        Map<YearMonth, BigDecimal> accumulatedGains,
        Map<YearMonth, BigDecimal> monthlyGains) {}

File: src/main/kotlin/org/moinex/model/dto/InvestmentPerformanceDTO.kt
/*
 * Filename: InvestmentPerformanceDTO.kt
 * Created on: February 17, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrate to Kotlin on 07/03/2026
 */

package org.moinex.model.dto

import java.math.BigDecimal
import java.time.YearMonth

data class InvestmentPerformanceDTO(
    val monthlyInvested: Map<YearMonth, BigDecimal>,
    val portfolioValues: Map<YearMonth, BigDecimal>,
    val accumulatedGains: Map<YearMonth, BigDecimal>,
    val monthlyGains: Map<YearMonth, BigDecimal>,
)
